chore: remove `r/gnoland/users` (#5194)
gnolang/gno: docs/resources/users-and-teams.md

# Users and Teams in gno.land

## Namespaces

> All addresses can deploy under their own address-based namespace:

Reviewer: Maybe

```suggestion
In Gno.land, anyone can deploy under their own address-based namespace:
```

```
gno.land/p/g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5/...
gno.land/r/g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5/...
```

The namespace is automatically derived from the deployer's address --
no registration is needed. These address-based namespaces allow for
permissionless deployment.

Username-based namespaces (e.g. `gno.land/r/myusername/myrealm`) are not

> currently supported and will be revisited via GovDAO governance.

Reviewer 2: I would modify this part like this:

```suggestion
Username-based namespaces (e.g. `gno.land/r/myusername/myrealm`) are not
currently supported but are planned to be introduced via GovDAO governance.
```

In Gno.land, users can register a unique username that:

> - Provides a more readable identity than a blockchain address

Reviewer 3: we can't just remove a single line, the whole feature is currently disabled.
